EKF/gps_checks.cpp

```diff
@@ -85,7 +85,13 @@ bool Ekf::collect_gps(const gps_message &gps)
 		_mag_strength_gps = 0.01f * get_mag_strength(lat, lon);
 
 		// request a reset of the yaw using the new declination
-		_mag_yaw_reset_req = true;
+		if (_params.mag_fusion_type == MAG_FUSE_TYPE_NONE) {
```

Comment:
I would really like to avoid having the yaw estimator reset silently hacked in the code through those `if (_params.mag_fusion_type == MAG_FUSE_TYPE_NONE) {` a bit everywhere.

In the context of this function, what's required is a yaw reset. Not specifically from a magnetometer or a yaw estimator implemented using an EKF GSF or a dual antenna GNSS receiver (btw, this is not handled yet), just a yaw reset. I think we could we simply set a generic variable `_yaw_reset_req = true` and let an external logic do the conversion to a specific reset depending on what's available and selected (we should also to a yaw reset using the solution of the yaw estimator if not `MAG_FUSE_TYPE_NONE` but that the mag data is not usable).